EZEE-3097 As a developer, I want to associate a site skeleton to a design (#1059)
ezsystems/developer-documentation: docs/guide/site_factory.md

    For full documentation on how Permissions work and how to set them up, see [the Permissions section](permissions.md).
    
    To learn how to use Site Factory, see [User documentation.](https://doc.ezplatform.com/projects/userguide/en/latest/site_organization/site_factory/#site-factory)
    
    ## Configure Site skeleton

>     The Site skeleton enables you to copy entire content structure of the site design to the defined Location.

Reviewer: ```suggestion
    The Site skeleton enables you to copy an entire content structure of the site design to the defined Location.
```

>     The Site skeleton enables you to copy entire content structure of the site design to the defined Location.

Reviewer: ```suggestion
    The Site skeleton enables you to copy entire an content structure of a site to the defined Location.
```

>     Site skeleton copying is one-off, it only happens during the site creation process.

Reviewer: ```suggestion
    Site skeleton copying is a one-off operation, it only happens during the site creation process.
```

>     After that, you cannot copy Site skeleton again, for example in the edit view.

Reviewer: ```suggestion
    After that, you cannot copy the Site skeleton again, for example in the edit view.
```

    
    You can create as many skeletons as you need and assign them to templates.
    Remember that one template can only have one Site skeleton assigned.
    

>     If the design does not have defined Site skeleton, a directory of the new site is created in a standard Site Factory process.

Reviewer: ```suggestion
    If the design does not have a defined Site skeleton, a directory of the new site is created in a standard Site Factory process.
```

    
    !!! caution "Update from v3.0"
    

>         To be able to create Location for the Site skeletons run `ezplatform:site-factory:create-site-skeletons-container` during update procedure.

Reviewer 2: ```suggestion
        To be able to create Location for the Site skeletons run `php ./bin/console ezplatform:site-factory:create-site-skeletons-container` during update procedure.
```

>         To be able to create Location for the Site skeletons run `php ./bin/console ezplatform:site-factory:create-site-skeletons-container` during update procedure.

Reviewer: ```suggestion
        To be able to create a Location for the Site skeletons, run `php ./bin/console ezplatform:site-factory:create-site-skeletons-container` during the update procedure.
```

>         Now, you should be able to see **Site skeletons** tab under **Admin** tab. The tab is only visible after update procedure.

Reviewer: ```suggestion
        Now, you should be able to see **Site skeletons** tab under **Admin** tab. The tab is only visible after the update procedure.
```

>     To define Site skeleton add a new configuration key to the template definition.

Reviewer: ```suggestion
    To define a Site skeleton, add a new configuration key to the template definition.
```

    This can be either a Location ID (e.g. `5966`), or a recommended remote Location ID (e.g. `3bed95afb1f8126f06a3c464e461e1ae66`).
    
    Add the configuration key to your template:
    
    ```yaml hl_lines="7 12"
    ez_platform_site_factory:
        templates:
            ez_site1:
                siteaccess_group: example_site_factory_group_1
                name: example_site_1
                thumbnail: /path/to/image/example-thumbnail_1.png
                site_skeleton_id: 5966
            ez_site2:
                siteaccess_group: example_site_factory_group_2
                name: example_site_2
                thumbnail: /path/to/image/example-thumbnail_2.png
                site_skeleton_remote_id: 3bed95afb1f8126f06a3c464e461e1ae66
    ```
    

>     Now, you can choose design with defined Site skeleton, and decide if you want to use its skeleton by toggling **Generate site using site skeleton**.

Reviewer: ```suggestion
    Now, you can choose a design with a defined Site skeleton, and decide if you want to use its skeleton by toggling **Generate site using site skeleton**.
```
